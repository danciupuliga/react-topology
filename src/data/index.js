export const graphql = [
  {
    index: 0,
    id: 'af6a5cd2-291f-490b-bf3b-141b010635db',
    name: 'frontend',
    status: 'ACTIVE',
    connections: ['aea06a05-830a-46d3-bdc1-9dcba97303de'],
    instances: [
      {
        id: 'f1fb3c1d-9e0e-4538-b2ad-1124bce2459e',
        status: 'RUNNING',
        healthy: 'UNKNOWN'
      },
      {
        id: 'c5c7ae33-cfe1-43cc-9e9b-6f453de3888d',
        status: 'FAILED',
        healthy: 'UNAVAILABLE'
      }
    ],
    instanceStatuses: [
      {
        status: 'RUNNING',
        count: 1
      },
      {
        status: 'FAILED',
        count: 1
      }
    ],
    instancesActive: true,
    instancesHealthy: {
      total: 2,
      healthy: 0
    },
    transitionalStatus: false,
    isConsul: false
  },
  {
    index: 1,
    id: 'af6a5cd2-291f-490b-bf3b-asdasads',
    name: 'GraphQL',
    status: 'ACTIVE',
    connections: ['af6a5cd2-291f-490b-bf3b-141b010635db'],
    instances: [
      {
        id: 'f1fb3c1d-9e0e-4538-b2ad-1124bce2459e',
        status: 'RUNNING',
        healthy: 'UNKNOWN'
      },
      {
        id: 'c5c7ae33-cfe1-43cc-9e9b-6f453de3888d',
        status: 'FAILED',
        healthy: 'UNAVAILABLE'
      }
    ],
    instanceStatuses: [
      {
        status: 'RUNNING',
        count: 2
      }
    ],
    instancesActive: true,
    instancesHealthy: {
      total: 2,
      healthy: 2
    },
    transitionalStatus: false,
    isConsul: false
  },
  {
    index: 2,
    id: 'af6a5cd2-291f-490b-bf3b-141b010635dbs',
    name: 'API',
    status: 'ACTIVE',
    connections: ['af6a5cd2-291f-490b-bf3b-asdasads'],
    instances: [
      {
        id: 'f1fb3c1d-9e0e-4538-b2ad-1124bce2459e',
        status: 'RUNNING',
        healthy: 'UNKNOWN'
      },
      {
        id: 'c5c7ae33-cfe1-43cc-9e9b-6f453de3888d',
        status: 'FAILED',
        healthy: 'UNAVAILABLE'
      },
      {
        id: 'c5c7ae33-cfe1-43cc-9e9b-6f453de3888d',
        status: 'FAILED',
        healthy: 'UNAVAILABLE'
      }
    ],
    instanceStatuses: [
      {
        status: 'RUNNING',
        count: 1
      },
      {
        status: 'FAILED',
        count: 1
      },
      {
        status: 'UNKNOWN',
        count: 1
      }
    ],
    instancesActive: true,
    instancesHealthy: {
      total: 3,
      healthy: 2
    },
    transitionalStatus: false,
    isConsul: false
  }
];
