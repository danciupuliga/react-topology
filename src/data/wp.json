[
  {
    "uuid": "081a792c-47e0-4439-924b-2efa9788ae9e",
    "id": "nginx",
    "name": "Nginx",
    "project": "e0ea0c02-55cc-45fe-8064-3e5176a59401",
    "instances": [""],
    "metrics": [
      {
        "name": "CPU",
        "value": "50%"
      },
      {
        "name": "Memory",
        "value": "20%"
      },
      {
        "name": "Network",
        "value": "2.9Kb/sec"
      }
    ],
    "connections": ["be227788-74f1-4e5b-a85f-b5c71cbae8d8"],
    "healthy": true,
    "datacentres": 1
  },
  {
    "uuid": "be227788-74f1-4e5b-a85f-b5c71cbae8d8",
    "id": "wordpress",
    "name": "Wordpress",
    "project": "e0ea0c02-55cc-45fe-8064-3e5176a59401",
    "instances": ["", ""],
    "metrics": [
      {
        "name": "CPU",
        "value": "50%"
      },
      {
        "name": "Memory",
        "value": "20%"
      },
      {
        "name": "Network",
        "value": "2.9Kb/sec"
      }
    ],
    "connections": [
      "6a0eee76-c019-413b-9d5f-44712b55b993",
      "6d31aff4-de1e-4042-a983-fbd23d5c530c",
      "4ee4103e-1a52-4099-a48e-01588f597c70"
    ],
    "healthy": true,
    "datacentres": 2
  },
  {
    "uuid": "6a0eee76-c019-413b-9d5f-44712b55b993",
    "id": "nfs",
    "name": "NFS",
    "project": "e0ea0c02-55cc-45fe-8064-3e5176a59401",
    "instances": ["", ""],
    "metrics": [
      {
        "name": "CPU",
        "value": "50%"
      },
      {
        "name": "Memory",
        "value": "20%"
      },
      {
        "name": "Network",
        "value": "2.9Kb/sec"
      }
    ],
    "healthy": true,
    "datacentres": 2
  },
  {
    "uuid": "6d31aff4-de1e-4042-a983-fbd23d5c530c",
    "id": "memcached",
    "name": "Memcached",
    "project": "e0ea0c02-55cc-45fe-8064-3e5176a59401",
    "instances": ["", ""],
    "metrics": [
      {
        "name": "CPU",
        "value": "50%"
      },
      {
        "name": "Memory",
        "value": "20%"
      },
      {
        "name": "Network",
        "value": "2.9Kb/sec"
      }
    ],
    "healthy": true,
    "datacentres": 2
  },
  {
    "uuid": "4ee4103e-1a52-4099-a48e-01588f597c70",
    "id": "percona",
    "name": "Percona",
    "project": "e0ea0c02-55cc-45fe-8064-3e5176a59401",
    "instances": ["", ""],
    "metrics": [
      {
        "name": "CPU",
        "value": "50%"
      },
      {
        "name": "Memory",
        "value": "20%"
      },
      {
        "name": "Network",
        "value": "2.9Kb/sec"
      }
    ],
    "healthy": true,
    "datacentres": 1
  },
  {
    "uuid": "97c68055-db88-45c9-ad49-f26da4264777",
    "id": "consul",
    "name": "Consul",
    "project": "e0ea0c02-55cc-45fe-8064-3e5176a59401",
    "instances": ["", ""],
    "isConsul": true,
    "metrics": [
      {
        "name": "CPU",
        "value": "50%"
      },
      {
        "name": "Memory",
        "value": "20%"
      },
      {
        "name": "Network",
        "value": "2.9Kb/sec"
      }
    ],
    "healthy": true,
    "datacentres": 2
  }
]
