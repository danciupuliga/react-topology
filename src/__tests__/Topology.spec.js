import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { mount, shallow, render } from 'enzyme';
import { graphql } from '../data';
import Topology from '../';

Enzyme.configure({ adapter: new Adapter() });

it('renders <Topology /> without throwing', () => {
  const tree = mount(<Topology services={graphql} />);
  expect(tree).toMatchSnapshot();
});

it('renders <Topology /> without throwing', () => {
  const tree = shallow(<Topology services={graphql} />);
  expect(tree).toMatchSnapshot();
});

it('renders <Topology /> without throwing', () => {
  const tree = render(<Topology services={graphql} />);
  expect(tree).toMatchSnapshot();
});

it('renders <Topology /> without throwing', () => {
  const fn = jest.fn();
  const tree = mount(<Topology services={graphql} onQuickActionsClick={fn} />);
  tree.find('.kMHeTL').first().simulate('click');
  expect(fn).toHaveBeenCalled()
  expect(tree).toMatchSnapshot();
});
