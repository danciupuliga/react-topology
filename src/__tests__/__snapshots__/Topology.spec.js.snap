// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Topology /> without throwing 1`] = `
<svg
  id="topology-svg"
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  onTouchCancel={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
>
  <g>
    <g
      transform="translate(490.90847863786803, 369.88822292304775)"
    >
      <rect
        className="sc-EHOje PMMGv"
        height={132}
        width={180}
        x={0}
        y={3}
      />
      <rect
        className="sc-ifAKCX hWxHou"
        height={132}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        width={180}
        x={0}
        y={0}
      />
      <g>
        <text
          className="sc-bZQynM hPybqV"
          onClick={[Function]}
          onKeyDown={[Function]}
          x={12}
          y={30}
        >
          frontend
        </text>
      </g>
      <g
        transform="translate(140, 0)"
      >
        <line
          className="sc-bxivhb gHpZWl"
          x1={0}
          x2={0}
          y1={0}
          y2={48}
        />
        <circle
          className="sc-iwsKbI hLQOJd"
          cx={20}
          cy={18}
          r={2}
        />
        <circle
          className="sc-iwsKbI hLQOJd"
          cx={20}
          cy={24}
          r={2}
        />
        <circle
          className="sc-iwsKbI hLQOJd"
          cx={20}
          cy={30}
          r={2}
        />
        <rect
          className="sc-dnqmqq kMHeTL"
          height={48}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={100}
          width={40}
        />
      </g>
      <g
        transform="translate(0, 48)"
      >
        <line
          className="sc-bxivhb gHpZWl"
          x1={0}
          x2={180}
          y1={0}
          y2={0}
        />
        <g
          transform="translate(12, 11)"
        >
          <g
            transform="translate(0, 0)"
          >
            <svg
              className="sc-VigVT bRjBVL healthy__StyledHealthyIcon-EmIZP iBQdnE"
              healthy="UNHEALTHY"
              height="18"
              viewBox="0 0 18 18"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                icon: state
              </title>
              <g
                fillRule="evenodd"
              >
                <circle
                  cx="9"
                  cy="9"
                  r="9"
                />
                <path
                  d="M9.477 6.603l-.522.443-.444-.443a2.056 2.056 0 1 0-2.908 2.909l3.352 3.352 3.412-3.352c.803-.804.863-2.106.06-2.91a2.105 2.105 0 0 0-2.95 0z"
                  fill="#FFF"
                />
              </g>
            </svg>
          </g>
          <g
            transform="translate(30, 4.5)"
          >
            <svg
              active={true}
              className="sc-gqjmRU gKImWu"
              consul={false}
              height="9"
              viewBox="0 0 18 9"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                icon: instances
              </title>
              <path
                d="M4.5 0a4.5 4.5 0 1 0 0 9 4.5 4.5 0 0 0 0-9m8.59 0c-.375 0-.739.047-1.09.12 2.184.456 3.818 2.239 3.818 4.38S14.184 8.424 12 8.88c.351.073.715.12 1.09.12C15.803 9 18 6.985 18 4.5S15.802 0 13.09 0M14 4.5C14 6.985 11.802 9 9.09 9c-.375 0-.739-.047-1.09-.12 2.184-.456 3.818-2.239 3.818-4.38S10.184.576 8 .12C8.351.047 8.715 0 9.09 0 11.803 0 14 2.015 14 4.5"
                fill="#494949"
                fillRule="evenodd"
              />
            </svg>
          </g>
          <text
            className="sc-htoDjs hKKxdB"
            x={54}
            y={14}
          >
            2 inst.
          </text>
          <g
            transform="translate(54, 36)"
          >
            <text
              className="sc-htoDjs hKKxdB"
            >
              1
            running
            </text>
            <text
              className="sc-htoDjs hKKxdB"
            >
              1
            failed
            </text>
          </g>
        </g>
      </g>
    </g>
    <g
      transform="translate(305.0915213621319, 540.1117770769522)"
    >
      <rect
        className="sc-EHOje iJhcIb"
        height={132}
        width={180}
        x={0}
        y={3}
      />
      <rect
        className="sc-ifAKCX fDXnCZ"
        height={132}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        width={180}
        x={0}
        y={0}
      />
      <g>
        <text
          className="sc-bZQynM hpEjYg"
          onClick={[Function]}
          onKeyDown={[Function]}
          x={12}
          y={30}
        >
          consul
        </text>
      </g>
      <g
        transform="translate(140, 0)"
      >
        <line
          className="sc-bxivhb ksbNYQ"
          x1={0}
          x2={0}
          y1={0}
          y2={48}
        />
        <circle
          className="sc-iwsKbI igBWAN"
          cx={20}
          cy={18}
          r={2}
        />
        <circle
          className="sc-iwsKbI igBWAN"
          cx={20}
          cy={24}
          r={2}
        />
        <circle
          className="sc-iwsKbI igBWAN"
          cx={20}
          cy={30}
          r={2}
        />
        <rect
          className="sc-dnqmqq kMHeTL"
          height={48}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={101}
          width={40}
        />
      </g>
      <g
        transform="translate(0, 48)"
      >
        <line
          className="sc-bxivhb ksbNYQ"
          x1={0}
          x2={180}
          y1={0}
          y2={0}
        />
        <g
          transform="translate(12, 11)"
        >
          <g
            transform="translate(0, 0)"
          >
            <svg
              className="sc-VigVT bRjBVL healthy__StyledHealthyIcon-EmIZP iBQdnE"
              healthy="HEALTHY"
              height="18"
              viewBox="0 0 18 18"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                icon: state
              </title>
              <g
                fillRule="evenodd"
              >
                <circle
                  cx="9"
                  cy="9"
                  r="9"
                />
                <path
                  d="M9.477 6.603l-.522.443-.444-.443a2.056 2.056 0 1 0-2.908 2.909l3.352 3.352 3.412-3.352c.803-.804.863-2.106.06-2.91a2.105 2.105 0 0 0-2.95 0z"
                  fill="#FFF"
                />
              </g>
            </svg>
          </g>
          <g
            transform="translate(30, 4.5)"
          >
            <svg
              active={true}
              className="sc-gqjmRU bhRpIl"
              consul={true}
              height="9"
              viewBox="0 0 18 9"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                icon: instances
              </title>
              <path
                d="M4.5 0a4.5 4.5 0 1 0 0 9 4.5 4.5 0 0 0 0-9m8.59 0c-.375 0-.739.047-1.09.12 2.184.456 3.818 2.239 3.818 4.38S14.184 8.424 12 8.88c.351.073.715.12 1.09.12C15.803 9 18 6.985 18 4.5S15.802 0 13.09 0M14 4.5C14 6.985 11.802 9 9.09 9c-.375 0-.739-.047-1.09-.12 2.184-.456 3.818-2.239 3.818-4.38S10.184.576 8 .12C8.351.047 8.715 0 9.09 0 11.803 0 14 2.015 14 4.5"
                fill="#494949"
                fillRule="evenodd"
              />
            </svg>
          </g>
          <text
            className="sc-htoDjs jJBSdK"
            x={54}
            y={14}
          >
            2 inst.
          </text>
          <g
            transform="translate(54, 36)"
          >
            <text
              className="sc-htoDjs jJBSdK"
            >
              1
            running
            </text>
            <text
              className="sc-htoDjs jJBSdK"
            >
              1
            running
            </text>
          </g>
        </g>
      </g>
    </g>
  </g>
  <g>
    <line
      className="sc-jTzLTM lotOOv"
      x1={483.0915213621319}
      x2={492.90847863786803}
      y1={542.1117770769522}
      y2={499.88822292304775}
    />
  </g>
  <g>
    <g
      transform="translate(492.90847863786803, 499.88822292304775) rotate(283.08870824588445)"
    >
      <circle
        className="sc-fjdhpX eCDqlH"
        cx={0}
        cy={0}
        r={9}
      />
      <line
        className="sc-jzJRlG ilIimZ"
        x1={-1}
        x2={2}
        y1={-3}
        y2={0}
      />
      <line
        className="sc-jzJRlG ilIimZ"
        x1={-1}
        x2={2}
        y1={3}
        y2={0}
      />
    </g>
  </g>
  <g />
  <g />
</svg>
`;
